remo/domain/dataset.py

```diff
@@ -316,37 +309,23 @@ def get_annotation_set(self, annotation_set_id: int = None) -> AnnotationSet:
         if not annotation_set_id:
             return self.default_annotation_set()
         
-        if annotation_set_id != -1:
-            annotation_set = self.sdk.get_annotation_set(annotation_set_id)
-            if annotation_set and annotation_set.dataset_id == self.id:
-                return annotation_set
-            else:
-                raise Exception('Annotation set with ID = {} is not part of dataset {}. You can check the list of annotation sets in your dataset using dataset.annotation_sets()'.format(annotation_set_id, self.name))
+        annotation_set = self.sdk.get_annotation_set(annotation_set_id)
+        if annotation_set and annotation_set.dataset_id == self.id:
+            return annotation_set
+        else:
+            raise Exception('Annotation set with ID = {} is not part of dataset {}. You can check the list of annotation sets in your dataset using dataset.annotation_sets()'.format(annotation_set_id, self.__str__()))
 
     def default_annotation_set(self) -> AnnotationSet:
         """
-        If a default annotation set exists, it returns that annotation set.
-        If a default annotation set doesn't exist, it sets the first annotation set to be default and returns that annotation set.
+        If the dataset has only one annotation set, it returns that annotation set. 
+        Otherwise, it raises an exception.
         """
-        if self._default_annotation_set_id:
-            return self.get_annotation_set(self._default_annotation_set_id)
-
         annotation_sets = self.annotation_sets()
-        if annotation_sets:
-            annotation_set = annotation_sets[0]
-            self._default_annotation_set_id = annotation_set.id
-            return annotation_set
-
-    def set_default_annotation_set(self, annotation_set_id: int):
-        """
-        Sets the default annotation set for a dataset. 
-        Important: default annotation sets are not stored in Remo, so every time a script runs 
-        the default annotation set will be assigned to the first annotation set that was created.
-
-        Args:
-            annotation_set_id: annotation set id
-        """
-        self._default_annotation_set_id = annotation_set_id
+        
+        if len(annotation_sets)>1:
+            raise Exception('Define which annotation set you want to use. ' + self.__str__() + ' has ' + str(len(annotation_sets)) + ' annotation sets. You can see them with my_dataset.annotation_sets()')
+                            
+        return annotation_set[0]
```

Comment:
@drewlr you didn't check case when there is no annotation sets, we will get an exception trying to retrieve first element from empty list

Reply:
@vmanilo good catch, thanks